lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -121,46 +112,16 @@ def get_plugin_status(self, plugin_class: Type[OpenSearchPlugin]) -> PluginState
                 return self.status(plugin)
         raise KeyError(f"Plugin manager did not find plugin: {plugin_class}")
 
-    def _extra_conf(self, plugin_data: Dict[str, Any]) -> Optional[Dict[str, Any]]:
-        """Returns the config from the relation data of the target plugin if applies."""
-        relation_name = plugin_data.get("relation")
-        relation = self._charm.model.get_relation(relation_name) if relation_name else None
-        # If the plugin depends on the relation, it must have at least one unit to be considered
-        # for enabling. Otherwise, relation.units == 0 means that the plugin has no remote units
-        # and the relation may be going away.
-        if relation and relation.units:
-            return {
-                **relation.data[relation.app],
-                **self._charm_config,
-                "opensearch-version": self._opensearch.version,
-            }
-        return {**self._charm_config, "opensearch-version": self._opensearch.version}
-
-    def check_plugin_manager_ready(self) -> bool:
-        """Checks if the plugin manager is ready to run."""
-        if not (deployment_desc := self._charm.opensearch_peer_cm.deployment_desc()):
-            return False
-
-        return (
-            self._charm.peers_data.get(Scope.APP, "security_index_initialised", False)
-            and self._charm.opensearch.is_node_up()
-            and len(
-                [
-                    node
-                    for node in self._charm._get_nodes(True)
-                    if node.app.id == deployment_desc.app.id
-                ]
-            )
-            == self._charm.app.planned_units()
-            and self._charm.health.get()
-            in [HealthColors.GREEN, HealthColors.YELLOW, HealthColors.IGNORE]
-        )
+    def is_ready_for_api(self) -> bool:
```

Comment:
More of a design opinion here, but these checks should probably be centralized at some point. They are invoked from outside this file, which means that it is a relevant check for other components.

Reply:
@zmraul can you give more precise references? AFAIU other places are asking if the opensearch is healthy, I am asking if it is responsive.